tools/release/adot-operator-images-mirror/docker_auth.go

```diff
@@ -1,22 +1,28 @@
 package main
 
 import (
-	"fmt"
-	"log"
+	"encoding/base64"
+	"encoding/json"
+	"io"
+	"strings"
 
-	docker "github.com/fsouza/go-dockerclient"
+	"github.com/docker/docker/api/types"
 )
 
-func getDockerCredentials(registry string) (*docker.AuthConfiguration, error) {
-	authOptions, err := docker.NewAuthConfigurationsFromDockerCfg()
+func getDockerCredentials(authToken string) (string, error) {
+	decoder := base64.NewDecoder(base64.StdEncoding, strings.NewReader(authToken))
+	credentials, err := io.ReadAll(decoder)
 	if err != nil {
-		log.Fatal(err)
+		return "", err
 	}
-
-	creds, ok := authOptions.Configs[registry]
-	if !ok {
-		return nil, fmt.Errorf("no auth found for %s", registry)
+	parts := strings.Split(string(credentials), ":")
```

Comment:
Any way we can get this as a struct instead of string split by : ?

Reply:
I don't believe so. That's just the way the authentication token is defined. See https://github.com/aws-observability/aws-otel-collector/pull/771#discussion_r760478940